packages/shared/src/features/briefing/hooks/useGenerateBrief.ts

```diff
@@ -0,0 +1,65 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useAuthContext } from '../../../contexts/AuthContext';
+import { gqlClient } from '../../../graphql/common';
+import type { ApiError, ApiErrorResult } from '../../../graphql/common';
+import type { Post } from '../../../graphql/posts';
+import { BriefingType, GENERATE_BRIEFING } from '../../../graphql/posts';
+import { generateQueryKey, RequestKey } from '../../../lib/query';
+
+interface UseGenerateBriefProps {
+  onSuccess?: (data: { id: string }) => Promise<void>;
+  onError?: (
+    error: ApiErrorResult<{
+      code: ApiError;
+      postId: string;
+      createdAt: string;
+    }>,
+  ) => void;
+}
+
+interface UseGenerateBriefReturn {
+  generateBrief: (params: { type?: BriefingType }) => void;
+  isGenerating: boolean;
+}
+
+export const useGenerateBrief = ({
```

Comment:
I don't think you need to create a new hook, you can use creator pattern with useMutation https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/1556185091/Data+fetching+mutations+and+optimistic+updates+with+typed+queries#Mutations

a little bit more versatile and you have all the options exposed from useMutation as normal.